File: Data Structures & Algorithms/max-water-container/submission-0.java
class Solution {
    public int maxArea(int[] height) {
        int lo = 0;
        int hi = height.length - 1;

        int champ = (hi - lo) * Math.min(height[lo], height[hi]);
        if (height[lo] > height[hi]) hi--;
        else lo++;

        while (lo < hi) {
            int current = (hi - lo) * Math.min(height[lo], height[hi]);
            if (current > champ) champ = current;
            if (height[lo] > height[hi]) hi--;
            else lo++;
        }

        return champ;
    }
}

File: Data Structures & Algorithms/jump-game-ii/submission-0.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        n = len(nums)
        goal = n - 1
        res = 0

        while goal != 0:
            champ = float('inf')
            for i in range(goal - 1, -1, -1):
                if nums[i] + i >= goal:
                    champ = min(i, champ)
            goal = champ
            res += 1
        
        return res
            
        





        